### ISSUE TITLE
Fix BinSkim errors 

### ISSUE THREAD

**Issue by jessamp** [microsoft/hermes-windows#90]
Hi, I have a request to fix the BinSkim errors with hermes. Looks like you'll need to change build flags to disable MD5 hashing and re-enable warnings 4146;4244;4267. When I run the task in our build pipeline, here's what binskim is showing us:

`hermes.dll: error BA2004: 'hermes.dll' is a native binary that directly compiles and links one or more object files which were hashed using an insecure checksum algorithm (MD5). MD5 is subject to collision attacks and its use can compromise supply chain integrity. Pass '/ZH:SHA_256' on the cl.exe command-line to enable secure source code hashing.

hermes.dll: error BA2007: 'hermes.dll' disables compiler warning(s) which are required by policy. A compiler warning is typically required if it has a high likelihood of flagging memory corruption, information disclosure, or double-free vulnerabilities. To resolve this issue, enable the indicated warning(s)`

**Comment by mganandraj** [comment]
Hi @jessamp, do you still need this to be looked at ?

**Comment by jessamp** [comment]
@mganandraj yes. We've worked around this by ignoring this error for now, but would prefer if the errors were fixed. Thanks!

**Comment by mganandraj** [comment]
Tudor checked in the following fix a while back to fix the object file checksum.
https://github.com/microsoft/hermes-windows/commit/bcda87c6179d33214412e0f3e1525fe5134cd3c0

but the compiler warnings are yet to be enabled.

**Comment by slobo80** [comment]
We also need this fix. Please prioritize if you can.

**Comment by vmoroz** [comment]
Currently hermes-windows repo has no BinSkim issue. Feel free to reopen this issue if you still see any.
